app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/RetrievedUserType.scala

```diff
@@ -29,27 +32,55 @@ object RetrievedUserType {
     ggCredId: GGCredId
   ) extends RetrievedUserType
 
+  object Individual{
+    implicit val format: OFormat[Individual] = derived.oformat()
+  }
+
   final case class Trust(sautr: SAUTR, email: Option[Email], ggCredId: GGCredId) extends RetrievedUserType
 
+  object Trust{
+    implicit val format: OFormat[Trust] = derived.oformat()
+  }
+
   final case class OrganisationUnregisteredTrust(email: Option[Email], ggCredId: GGCredId) extends RetrievedUserType
 
+  object OrganisationUnregisteredTrust{
+    implicit val format: OFormat[OrganisationUnregisteredTrust] = derived.oformat()
+  }
+
   final case class IndividualWithInsufficientConfidenceLevel(
     nino: Option[NINO],
     sautr: Option[SAUTR],
     email: Option[Email],
     ggCredId: GGCredId
   ) extends RetrievedUserType
 
+  object IndividualWithInsufficientConfidenceLevel {
+    implicit val format: OFormat[IndividualWithInsufficientConfidenceLevel] = derived.oformat()
+  }
+
   final case class Subscribed(
     cgtReference: CgtReference,
     ggCredId: GGCredId
   ) extends RetrievedUserType
 
+  object Subscribed{
+    implicit val format: OFormat[Subscribed] = derived.oformat()
```

Comment:
you don't need to use the derived instance here: you can just use the straightforward `Json.format` - the dervied instances are useful when you want to get a format instance for a whole sealed trait and you can't be bothered to write a big pattern match. 
This is why the `AuthenticatedActionWithRetrievedDataSpec` was failing for you. Do you even need the format instances? These are never serialised except for in that spec.

Reply:
No it even failed when using the Json.format. I switched to derived instances as I thought it would be clever enough to generate the correct JSON but it didn't. I can take these out now as I changed the tests to manually construct the JSON

Reply:
the derived format will include the case class/object name in the JSON which isn't normally there when you use `Json.format`

Reply:
The compiler complained that it could not find codecs for the these types in the test. That's why I added them. The tests seemed broken before the upgrade but it wasn't getting picked up.

Reply:
> the derived format will include the case class/object name in the JSON which isn't normally there when you use `Json.format`

it didn't do that when I tried  - you can try and change the code back and see if it works for you